### DISCUSSION TITLE
Migrating did:webs

### DISCUSSION THREAD

**Discussion by dholms** [bluesky-social/atproto#1874] [General Discussion]
We recently internally federated our PDS and migrated all of our accounts. 

Unfortunately, for `did:web`s, we are unable to update the signing key and pds endpoint in your DID document. In order to keep using your account, please update you did document accordingly:

`did:web:dead10ck.dev`
- pds endpoint: `https://puffball.us-east.host.bsky.network`
- signing key: `did:key:zQ3shmGvLE3VyF2YnzWVsxAJf6mEiWDK5LaCrPSZpZCohikwb`

`did:web:lily.pet`
- pds: endpoint: `https://inkcap.us-east.host.bsky.network`
- signing key: `did:key:zQ3shZRdzg3LWKKNpL59kXvvFZVVB3c5cdX35gXcFaY6HeZ7g`

`did:web:retr0.id`
- pds: endpoint: `https://oyster.us-east.host.bsky.network`
- signing key: `did:key:zQ3shPp74h4BoAZ3bEBLfBBGiohF9EsUks1CYr79nSLHu3zjN`

`did:web:retr0.id/stuff/didweb` (INVALID/DISALLOWED DID)
- pds: endpoint: `https://enoki.us-east.host.bsky.network`
- signing key: `did:key:zQ3sha6BmsvFADW8CwMaV8vEfTFLbPb2ceZwAmFTgcnQRff2z`

`did:web:yamarten.github.io`
- pds: endpoint: `https://porcini.us-east.host.bsky.network`
- signing key: `did:key:zQ3shR87pRk3R8Vf5QVFfwCMoRLB8TgfdbTBK3kCktXeKG4AU`

**Comment by DavidBuchanan314** [comment]
By the way, my `did:web:retr0.id/stuff/didweb` DID is "unsupported" according to the [docs](https://atproto.com/specs/did).

> In the context of atproto, only hostname-level did:web DIDs are supported: path-based DIDs are not supported. 

I can log in to the account, but it's very broken and I've never been able to create any records etc., and I assume the repo is empty.

**Reply by bnewbold** [reply]
Yup! Was just looking at that and was initially very worried it was somehow your primary account. It shouldn't be possible to create accounts like that any more, and I think it never would have propagated through the relay (different codebase from PDS).

We will likely need to delete or remove that account at some point. If there is any special cleanup or "in memoria" you would like, let us know.

**Reply by DavidBuchanan314** [reply]
I'm ok for it to be deleted unceremoniously, although it would be cool if the string got used in a regression test somewhere :)

**Reply by mackuba** [reply]
Btw it's currently showing this on bsky.app if I try to open that profile ;)

<img width="354" alt="Screen Shot 2024-03-06 at 23 53 05" src="https://github.com/bluesky-social/atproto/assets/28465/d709f465-7d51-4624-8736-5e04300a57bd">

**Comment by dead10ck** [comment]
I tried using this key and endpoint, and I still get an Internal Error banner when I try to log in with the app. Am I doing something wrong here?

```sh
❯ : http get https://dead10ck.dev/.well-known/did.json -r | jq
{
  "@context": [
    "https://www.w3.org/ns/did/v1",
    "https://w3id.org/security/suites/secp256k1-2019/v1"
  ],
  "id": "did:web:dead10ck.dev",
  "alsoKnownAs": [
    "at://dead10ck.dev"
  ],
  "verificationMethod": [
    {
      "id": "did:web:dead10ck.dev#atproto",
      "type": "Multikey",
      "controller": "did:web:dead10ck.dev",
      "publicKeyMultibase": "zQ3shmGvLE3VyF2YnzWVsxAJf6mEiWDK5LaCrPSZpZCohikwb"
    }
  ],
  "service": [
    {
      "id": "#atproto_pds",
      "type": "AtprotoPersonalDataServer",
      "serviceEndpoint": "https://puffball.us-east.host.bsky.network"
    }
  ]
}
```

**Reply by dholms** [reply]
Your did doc looks good & everything seems to look good on our end.

Do you mind trying again? There may have been something sticking around in a cache somewhere 🤔

**Reply by dead10ck** [reply]
There were go, it's working now! Thanks all
